src/main/java/de/unistuttgart/bugfinder/gameresult/GameResultService.java

```diff
@@ -0,0 +1,195 @@
+package de.unistuttgart.bugfinder.gameresult;
+
+import de.unistuttgart.bugfinder.code.Code;
+import de.unistuttgart.bugfinder.configuration.Configuration;
+import de.unistuttgart.bugfinder.configuration.ConfigurationRepository;
+import de.unistuttgart.bugfinder.configuration.ConfigurationService;
+import de.unistuttgart.bugfinder.solution.Solution;
+import de.unistuttgart.bugfinder.solution.SolutionDTO;
+import de.unistuttgart.bugfinder.solution.SolutionRepository;
+import de.unistuttgart.bugfinder.solution.bug.Bug;
+import de.unistuttgart.bugfinder.solution.bug.BugDTO;
+import feign.FeignException;
+import java.util.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+@Slf4j
+public class GameResultService {
+
+  static final long MAX_SCORE = 100l;
+
+  @Autowired
+  private ConfigurationRepository configurationRepository;
+
+  @Autowired
+  private ConfigurationService configurationService;
+
+  @Autowired
+  private SolutionRepository solutionRepository;
+
+  @Autowired
+  private ResultClient resultClient;
+
+  public void saveGameResult(final GameResultDTO gameResultDTO, final String userId) {
+    final OverworldResultDTO resultDTO = new OverworldResultDTO(
+      "BUGFINDER",
```

Comment:
The last thing I have to criticize is that the `OverworldResultDTO` is responsible for ensuring the string is `BUGFINDER` in **all** cases, and that callers should not need to guess what string is expected.
So, either ensure this in the constructor, or if you know of any annotation for it use that.